submissions/YarHadi/task_js_oop/app.js

```diff
@@ -0,0 +1,133 @@
+const startPlayerX = 200,
+  startPlayerY = 390,
+  cellWidth = 101,
+  cellHeight = 83,
+  leftBorder = 0,
+  rightBorder = 303,
+  topBorder = 0;
+
+// Enemies our player must avoid
+
+var Enemy = function (x, y) {
+  // Variables applied to each of our instances go here,
+  // we've provided one for you to get started
+
+  this.x = x;
+  this.y = y;
+  this.speed = Math.random() * 200 + 200;
+
+  // The image/sprite for our enemies, this uses
+  // a helper we've provided to easily load images
+
+  this.sprite = "images/enemy-bug.png";
+};
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+
+Enemy.prototype.update = function (dt) {
+    
+  // You should multiply any movement by the dt parameter
+  // which will ensure the game runs at the same speed for
+  // all computers.
+
+  this.x += this.speed * dt;
+  if (this.x > ctx.canvas.width) {
+    this.x = 0;
+  }
+
+  if (this.collision()) {
+    player.resetPlayer();
+  }
+};
+
+Enemy.prototype.collision = function () {
+  if (Math.abs(this.x - player.x) < 40 && Math.abs(this.y - player.y) < 40) {
+    return 1;
+  }
+};
+
+// Draw the enemy on the screen, required method for game
+
+Enemy.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+
+var Player = function (x, y) {
+  this.resetPlayer();
+  this.sprite = "images/char-boy.png";
+};
+
+Player.prototype.resetPlayer = function () {
+  this.x = startPlayerX;
+  this.y = startPlayerY;
+};
+
+Player.prototype.update = function () {
+  if (this.y < topBorder) {
+    this.resetPlayer();
+  }
+};
+
+Player.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Player.prototype.handleInput = function (key) {
+  switch (key) {
+    case "left":
+      if (this.x > leftBorder) {
+        this.x -= cellWidth;
+      }
+      break;
+    case "right":
+      if (this.x < rightBorder) {
+        this.x += cellWidth;
+      }
+      break;
+    case "up":
+      this.y -= cellHeight;
+      break;
+    case "down":
+      if (this.y < startPlayerY) {
+        this.y += cellHeight;
+      }
+      break;
+  }
+};
+
+// Now instantiate your objects.
+// Place all enemy objects in an array called allEnemies
+// Place the player object in a variable called player
+
+const player = new Player();
+
+const allEnemies = [];
+
+//Creating enemies
+
+function createEnemies() {
+  const posEnemY = [60, 140, 230];
+  for (let y of posEnemY) {
+    allEnemies.push(new Enemy(0, y));
+  }
```

Comment:
(1) what are 60, 140, 230?
See section 3 of the requirements linked by bot.
Prefer calculations where there is any correlation.

(2) It is a classical use case for `Array#map` where you would convert a list of mystery numbers into a list of instances of class `Enemy`. You will have a ingle line of code, and, probably, having a function will be a bit overly unless you use 1+ times across the code.